Apps/phgsgmail/gsgmail_connector.py

```diff
@@ -269,23 +367,23 @@ def _handle_run_query(self, param):
         query_string = ""
```

Comment:
you can simplify the query construction logic and remove all the duplicate code by using a `.join`, something like:

```
query_dict = {
    'label': param.get('label'),
    'subject': param.get('subject'),
    ...
}
query_string = ' '.join('{}:{}'.format(key, value) for key, value in query_dict if value is not None)
```

not sure if we still need the `_handle_py_ver_compat_for_input_str`? since I think we only need to support py3 now but you can also add that to the list comprehension in the join if we do

Reply:
i guess body doesn't have an associated query variable in the string so I refactored the others and kept that as is, since the logic to build body into that comprehension would have been ugly